app/lib/frontend/search_service.dart

```diff
@@ -62,11 +70,65 @@ class SearchService {
   /// The Custom Search client API stub.
   final customsearch.CustomsearchApi csearch;
 
-  SearchService(this.httpClient, this.csearch);
+  /// The HTTP client used for making calls to our search service.
+  final http.Client searchServiceClient;
+
+  SearchService(this.httpClient, this.csearch, this.searchServiceClient);
 
   /// Search for packes using [queryText], starting at offset [offset] returning
   /// max [numResults].
-  Future<SearchResultPage> search(SearchQuery query) async {
+  Future<SearchResultPage> search(SearchQuery query, bool useService) async {
+    if (useService) {
+      try {
+        final SearchResultPage page = await _searchService(query);
+        if (page != null) return page;
+      } catch (e, st) {
+        _logger.severe('Unable to call search service.', e, st);
+      }
+    }
+    return _searchCSE(query);
+  }
+
+  Future<SearchResultPage> _searchService(SearchQuery query) async {
+    final search_service.PackageQuery packageQuery =
+        new search_service.PackageQuery(query.text,
+            type: query.type, offset: query.offset, limit: query.limit);
+    final Uri serviceUrl = new Uri(
+      scheme: 'https',
+      host: activeConfiguration.searchServiceHost,
+      path: '/search',
+      queryParameters: packageQuery.toServiceQueryParameters(),
+    );
+    final http.Response response = await searchServiceClient.get(serviceUrl);
+    if (response.statusCode == search_service.searchIndexNotReadyCode) {
+      // Search request while service instance is updating index.
```

Comment:
We might be better off checking for three exit codes: 
  * the positive one if the response was correctly satisfied
  * the negative one where the service is up & running but is building the very first index
  * everything else, e.g. 500s, internal errors, ... (should log these as errors)